core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -272,6 +299,16 @@ public DefaultMuleEvent(MuleEvent rewriteEvent, FlowConstruct flowConstruct)
             rewriteEvent.isSynchronous());
     }
 
+    /**
+     * Copy constructor used when ReplyToHandler instance needs switching out
+     * @param rewriteEvent
+     * @param replyToHandler
+     */
+    public DefaultMuleEvent(MuleEvent rewriteEvent, ReplyToHandler replyToHandler)
```

Comment:
It was not my intention to add that as part of this PR but just to make a point. It would be good to do it on 3.7